### PR TITLE
implement verify method

### PR DESCRIPTION
**Pull request by tkfinitefield**
Fixed issue: Auth#verifyIdToken always throws unimplemented error #20

### CODE REVIEW

**Comment by davidcz95** [comment]
Will it be merge in next release ? Just tried it from commit ref and it is working!

**Comment by rrousselGit** [comment]
I'm in holiday at the moment. I'll look at this when I'm back from Japan.

**Comment by davidcz95** [comment]
Suure no worries. Have fun there! 🇯🇵

**Comment by KKimj** [comment]
Hi @rrousselGit , I hope you're enjoying your trip to Japan! Just wanted to gently remind you about this, awaiting your review when you have a chance.

**Comment by rrousselGit** [comment]
Looks like this needs testing. My apologies, I should've raised this sooner.

I'll review other PRs and fix other bugs in the meantime. If it's still around by then, I'll consider adding tests for this myself :)

**Review by rrousselGit — CHANGES_REQUESTED** [review]
Cf above (testing)

File: packages/dart_firebase_admin/lib/src/utils/jwt.dart [hunk 1]
```diff
@@ -4,16 +4,19 @@ import 'package:dart_jsonwebtoken/dart_jsonwebtoken.dart';
...
 class EmulatorSignatureVerifier implements SignatureVerifier {
   @override
   Future<void> verify(String token) async {
     // Signature checks skipped for emulator; no need to fetch public keys.
     try {
-      return await verifyJwtSignature(
-        token,
-        SecretKey(''),
-      );
+      JWT.decode(token);
```
Comment (rrousselGit) [inline packages/dart_firebase_admin/lib/src/utils/jwt.dart:19]: Why this change?

File: packages/dart_firebase_admin/lib/src/utils/jwt.dart [hunk 2]
```diff
@@ -96,9 +99,51 @@ class PublicKeySignatureVerifier implements SignatureVerifier {
...
+        token,
+        RSAPublicKey.cert(publicKey),
+      );
+    } on JWTExpiredException {
+      throw JwtError(
+        JwtErrorCode.tokenExpired,
+        'The provided token has expired. Get a fresh token from your client app and try again.',
+      );
+    } on JWTException catch (e) {
+      if (e.message.startsWith(jwtCallbackErrorPrefix)) {
```
Comment (rrousselGit) [inline packages/dart_firebase_admin/lib/src/utils/jwt.dart:133]: We shouldn't depend on the english message.

Comment (KKimj) [reply packages/dart_firebase_admin/lib/src/utils/jwt.dart:133]: And then, how about simplifying this error handling for now?

Just like:

```dart
...

} on JWTExpiredException {
      throw JwtError(
        JwtErrorCode.tokenExpired,
        'The provided token has expired. Get a fresh token from your client app and try again.',
      );
    } on JWTException catch (e) {
    // Just throw JwtErrorCode.unknown for now.
    throw JwtError(JwtErrorCode.unknown, e.message);
}

...


enum JwtErrorCode {
  invalidArgument('invalid-argument'),
   ...

   // add this enum field
  unknown('unknown');

  ...
}


```

Comment (rrousselGit) [reply packages/dart_firebase_admin/lib/src/utils/jwt.dart:133]: Sure. I shouldn't have done that indeed.

Comment (KKimj) [reply packages/dart_firebase_admin/lib/src/utils/jwt.dart:133]: @tkfinitefield 

Hello! 
I would appreciate it if you could take a moment to check these comments and share your thoughts.

**Comment by rrousselGit** [comment]
This has been sorted in a different PR. Thanks for your work though and sorry for the slow review!
